gordo_components/builder/logsender.py

```diff
@@ -0,0 +1,127 @@
+from typing import Callable
+
+from uuid import uuid4
+import logging
+import warnings
+
+from gordo_components.builder.azure_utils import (
+    get_mlflow_client,
+    get_run_id,
+    get_batch_kwargs,
+    get_spauth_kwargs,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class LogSender:
+    """
+    External logging wrapper
+
+    Parameters
+    ----------
+    func: Callable
+        Then function whose dictionary of items to capture for logging.
+    name: str
+        The name of the log group to log to (e.g. a model name).
+    model_key: str
+        Unique ID of logging run.
+    workspace_kwargs: dict
+        AzureML Workspace keyword arguments to use for remote MLFlow tracking.
+        Example::
+
+            `{
+                 "subscription_id":<value>,
+                 "resource_group":<value>,
+                 "workspace_name":<value>
+             }`
+    service_principal_kwargs: dict
+        AzureML ServicePrincipalAuthentication keyword arguments. An empty dict or None
+        will result in interactive authentication.
+        Example::
+
+            `{
+                 "tenant_id":<value>,
+                 "service_principal_id":<value>,
+                 "service_principal_password":<value>
+             }`
+
+    Example
+    -------
+
+    >>> def func(a, b):
+    ...     metadata = {"a": a, "b": b}
+    ...     return a+b, metadata
+    ...
+    >>> def execute_and_log():
+    ...     # Pass empty workspace config for InteractiveLoginAuthentication
+    ...     func = LogSender(func, "log_group", "unique_key", {})
+    ...     result, metadata = func(1, 2)
+    ...     print(result)
+    ...
+    >>> execute_and_log() # doctest: +SKIP
+    3
+    """
+
+    def __init__(
+        self,
+        func: Callable = None,
+        name: str = "",
+        model_key: str = "",
+        workspace_kwargs: dict = {
+            "subscription_id": "019958ea-fe2c-4e14-bbd9-0d2db8ed7cfc",
```

Comment:
This seems weird to have here, I don't know much about it though...but _very_ strange to have this in an open source project.